### DISCUSSION TITLE
Manual tasks only

### DISCUSSION THREAD

**Discussion by vg67000** [immich-app/immich#12686] [General]
Hello,

Thank you for your great work in developping immich, I'm quite new to this app but I really appreciate it.
I installed it on a small NUC that I had and the CPU is not very powerfull. My problem is that when I upload multiple photos/videos, all tasks are running at the same time, making it very slow to run the tasks. So what I usually do is pausing all running tasks and execute them one by one (first generate all thumbnails, then smart search, then face detection, then facial recognition). I think it'll be nice to have the option of not running the tasks automatically. For example, when I upload new pictures, I'd like to generate the thumbnails automatically and do the smart search, but I don't want to run face detecion/recognition automatically, and I'd like to be able to run it manually later, without deactivating the whole functionnality.
I've searched in other discussions and didn't find anything like it, and didn't find any option like this in the software.

Regards

**Comment by radh21301** [comment]
One easy way to do this is disabling the 'Face recognition' and 'Smart search' in the settings. When you want to run, just enable them - and run 'Missing' job for both smart search and face detection.It will only run of the images that have not been processed yet. So, you are not doing anything inefficient.
Once done, disable them again.

Note: When you run face detection job, facial recognition is automatically queued. So, you dont have to run the face recognition job

**Reply by vg67000** [reply]
I found this trick to, but it would be better to avoid going to the settings all the time to activate/deactive the functionality.

Also, facial recognition is automatically queued & launched, and that's my problem. Both are using a lot of resources on the computer, and it makes both jobs very slow when running at the same time.

**Reply by radh21301** [reply]
Reduce the concurrency to 1 for all the jobs. That should do the trick?

Lot of people are running it on a raspberry pi, and it seems to be fine. Wonder if your nuc is weaker than those.

Also - these things should not be a problem with day to day use right? I mean, one might upload say 20 pics a day on average (based on my average)..and occasionally have a lot of pics when they go on a trip or something. This should not really cause any issues. And I think running things with a concurrency of 1 should work just fine.

Honestly, you keep looking at the clock to see if things are done -> only the first time you set immich up..Once you have it going, you never look at the jobs page at all. So, I wouldnt worry if it takes too long to process the initial job.
